engine/src/main/java/org/terasology/engine/rendering/dag/RenderTaskListGenerator.java

```diff
@@ -167,23 +167,28 @@ public List<RenderPipelineTask> generateFrom(List<Node> orderedNodes) {
         long endTimeInNanoSeconds = System.nanoTime();
 
         // if (logger.isDebugEnabled()) {
-            logger.debug("===== INTERMEDIATE RENDERER LIST =========================");
+            logger.atDebug().log("===== INTERMEDIATE RENDERER LIST =========================");
             logIntermediateRendererListForDebugging(orderedNodes);
-            logger.debug("===== RENDERER TASK LIST =================================");
+            logger.atDebug().log("===== RENDERER TASK LIST =================================");
             logList(taskList);
-            logger.debug("----------------------------------------------------------");
-            logger.debug("Task list generated in {} ms", String.format("%.3f", (endTimeInNanoSeconds - startTimeInNanoSeconds) / 1000000f));
-            logger.debug("{} nodes, {} enabled - {} tasks (excluding marker tasks) out of {} potential tasks.",
-                    nodeList.size(), enabledNodes, taskList.size() - enabledNodes, potentialTasks);
-            logger.debug("----------------------------------------------------------");
+            logger.atDebug().log("----------------------------------------------------------");
+            logger.atDebug().addArgument(() -> String.format("%.3f", (endTimeInNanoSeconds - startTimeInNanoSeconds) / 1000000f)).
+                    log("Task list generated in {} ms");
+            logger.atDebug().
+                    addArgument(() -> nodeList.size()).
+                    addArgument(enabledNodes).
+                    addArgument(taskList.size() - enabledNodes).
+                    addArgument(potentialTasks).
+                    log("{} nodes, {} enabled - {} tasks (excluding marker tasks) out of {} potential tasks.");
+            logger.atDebug().log("----------------------------------------------------------");
         // }
 
         return taskList;
     }
 
     private void logList(List<?> list) {
         for (Object object : list) {
-            logger.debug("{}", object);
+            logger.atDebug().addArgument(object).log("{}");
```

Comment:
why do we need fluent API here when there's no method call involved...?

Reply:
what do you mean by "need". fluent api is not painful, and not discouraged :) additionally, this is a loop over a list.

Reply:
It's slightly more painful to read, so I would slightly discourage using it except for cases where the added verbosity makes sense due to some kind of need. if there's no need I'd always go for the easier to read version.

Reply:
easier to read is in the eye of the one reading. for me fluent is now easier to read :) but i understand, having not written fluent for you it looks stranger. but here, the main reason is that it loops over a list of objects.

Reply:
it may loop over a list of objects, but there's no additional calculation involved that is performed for every object...
did PMD complain about this?